piet/src/samples/mod.rs

```diff
@@ -76,53 +77,67 @@ struct Args {
 /// PNG to the path.
 ///
 /// The `prefix` argument is used for the file names of failure cases.
-pub fn samples_main(f: fn(usize, &Path) -> Result<(), BoxErr>, prefix: &str) -> Result<(), BoxErr> {
-    let args = match Args::from_env() {
-        Ok(args) => args,
-        Err(e) => {
+pub fn samples_main(f: fn(usize, &Path) -> Result<(), BoxErr>, prefix: &str) -> ! {
+    let inner = move || -> Result<(), BoxErr> {
+        let args = Args::from_env()?;
```

Comment:
So the fact that we create an execute a closure means we can collect all error handling in one place (at the bottom of this function). Since the help is printed regardless of error, current behaviour is maintained, but we also print the help for any other error the user encounters. This is the primary motivation for this patch: the confusion I saw on Zulip was (I think) using a number outside the valid range for picture number. Printing the help in that case should make it easier for the user to understand the problem without needing to ask.